acinclude/printf.m4

```diff
@@ -0,0 +1,33 @@
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+dnl check which printf code to use for a type
+dnl $1 - the type being checked
+dnl $2 - the PRIxxx macro to define
+dnl $3 - a list of possible codes
+AC_DEFUN([SQUID_CHECK_PRINTF_CODES],[
+  AH_TEMPLATE([$2],[printf display of $1])
+  AC_CACHE_CHECK([which printf code displays $1],
+                 [squid_cv_printf_$2],[
+    SQUID_STATE_SAVE($2)
+    CXXFLAGS="$CXXFLAGS -Werror=format"
```

Comment:
> There are some weird things going on with flags set (or not) by some tests

Yes, sounds like it.

> For now I think this is "good enough" and removal can wait on future work cleaning up that mess.

Since this addition is a temporary and _problematic_ workaround for other (yet-to-be-identified) ./configure bugs, it should be marked as such with a source code comment. For example,

```suggestion
    CXXFLAGS="$CXXFLAGS -Werror=format" # XXX: This should not be needed for most builds (and breaks the test for some others)
```